specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+achieved using role aggregation; on systems using Kubernetes RBAC, resources
+which wish to participate in Addressable resolution are expected to provide the
+following `ClusterRole`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <component>-addressable-resolver
+  labels:
+    duck.knative.dev/addressable: "true"
+rules:
+  - apiGroups: [<resource apiGroup>]
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
+support.
+
+<!-- TODO: define aggregated API roles
+
+Ref:
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-channelable-manipulator-clusterrole
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-addressable-resolver-clusterrole
+-
+-->
+
+## Error Signalling
+
+See [the Knative common condition guidance](../common/error-signalling) for how
+resource errors are signalled to the user.
+
+## Resource Lifecycle
+
+### Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it MUST have a
+`status.address.url` field) which can receive, store, and forward events to
+multiple recipients based on a set of attribute filters (Triggers). Triggers are
+associated with a Broker based on the `spec.broker` field on the Trigger; it is
+expected that the controller for a Broker will also control the associated
+Triggers. When the Broker's `Ready` condition is `true`, the Broker MUST provide
+a `status.address.url` which accepts all valid CloudEvents and MUST attempt to
+forward the received events for filtering to each associated Trigger whose
+`Ready` condition is `true`. As described in the
+[Trigger Lifecycle](#trigger-lifecycle) section, a Broker MAY forward events to
+an associated Trigger which which does not currently have a `true` `Ready`
+condition, including events received by the Broker before the Trigger was
+created.
+
+The annotation `eventing.knative.dev/broker.class` MAY be used to select a
+particular implementation of a Broker. When a Broker is created, the
+`eventing.knative.dev/broker.class` annotation and the `spec.config` field MUST
+be populated (`spec.config` MAY be an empty object) to indicate which of several
+possible Broker implementations to use. It is RECOMMENDED to default the
+`eventing.knative.dev/broker.class` field on creation if it is unpopulated. Once
+created, both fields MUST be immutable; the Broker MUST be deleted and
+re-created to change the implementation class or `spec.config`. This pattern is
+chosen to make it clear that changing the implementation class or `spec.config`
+is not an atomic operation and that any implementation would be likely to result
+in event loss during the transition.
+
+### Trigger Lifecycle
+
+The lifecycle of a Trigger is independent of the Broker it refers to in its
```

Comment:
This statement isn't true given that it's Ready condition isn't True until the Broker is ready to send it events. See my comment below on this